pkg/rehearse/executor.go

```diff
@@ -0,0 +1,147 @@
+package rehearse
```

Comment:
I don't think this needs to live in a separate file, it's unnecessarily fragmented. It still handles jobs.